pkg/resources/materialized_view.go

```diff
@@ -230,76 +176,86 @@ func ReadMaterializedView(d *schema.ResourceData, meta interface{}) error {
 		return err
 	}
 
-	return d.Set("database", v.DatabaseName.String)
+	return nil
 }
 
 // UpdateMaterializedView implements schema.UpdateFunc.
 func UpdateMaterializedView(d *schema.ResourceData, meta interface{}) error {
-	mvid, err := materializedViewIDFromString(d.Id())
-	if err != nil {
-		return err
-	}
-
-	dbName := mvid.DatabaseName
-	schema := mvid.SchemaName
-	view := mvid.ViewName
-
-	builder := snowflake.NewMaterializedViewBuilder(view).WithDB(dbName).WithSchema(schema)
-
 	db := meta.(*sql.DB)
+	ctx := context.Background()
+	client := sdk.NewClientFromDB(db)
+	id := helpers.DecodeSnowflakeID(d.Id()).(sdk.SchemaObjectIdentifier)
+
 	if d.HasChange("name") {
```

Comment:
Added `id = newId` to prevent what you described. Also changed one prop in the rename test to check it. Good catch! I wonder how many other resources have this problem -  I will add this to one of the tasks we have with common errors in the repo.